app/services/common_platform/api/defendant_finder.rb

```diff
@@ -15,30 +14,50 @@ def call
     private
 
       def common_platform_defendant
-        @common_platform_defendant ||= common_platform_prosecution_case&.defendants&.find { |d| d.id.eql?(defendant_id) }
+        return @common_platform_defendant if @common_platform_defendant
+        return unless prosecution_case_urns
+
+        prosecution_case_urns.each do |urn|
+          prosecution_case = common_platform_prosecution_case(urn)
+
+          next unless prosecution_case
+
+          @common_platform_defendant ||= prosecution_case&.defendants&.find { |d| d.id.eql?(defendant_id) }
+
+          break if @common_platform_defendant
+        end
+
+        @common_platform_defendant
```

Comment:
Minor comment:
what about return nil here (instead of relying on instance variable default)
and direct return
`return @common_platform_defendant if @common_platform_defendant`
on line 27?
I think it's more clear: once you find a @common_platform_defendant you return!

Reply:
Nice! That's much clearer.